decorator.rb

```diff
@@ -0,0 +1,26 @@
+require './nameable'
+
+class Decorator < Nameable
+  attr_accessor :nameable
+
+  def initialize(nameable)
+    super()
+    @nameable = nameable
+  end
+
+  def correct_name
+    @nameable.correct_name
+  end
+end
+
+class CapitalizeDecorator < Decorator
+  def correct_name
+    @nameable.correct_name.capitalize
+  end
+end
+
+class TrimmerDecorator < Decorator
+  def correct_name
+    @nameable.correct_name.length > 10 ? @nameable.correct_name.slice(0, 10) : @nameable.correct_name
+  end
+end
```

Comment:
- [Optional] Kindly, move these class to their own file to have clean code 👍

Reply:
well noted @binhussen. I will do that. Thanks a lot